aidb/query/query.py

```diff
@@ -153,36 +176,29 @@ def _get_sympify_form(self, node, predicate_count, predicate_mappings):
             isinstance(node, exp.LTE) or \
             isinstance(node, exp.EQ) or \
             isinstance(node, exp.Like) or \
-            isinstance(node, exp.NEQ):
-      # TODO: chained comparison operators not supported
-      assert "this" in node.args and "expression" in node.args
+            isinstance(node, exp.NEQ) or \
+            isinstance(node, exp.In) or \
+            isinstance(node, exp.Not) or \
+            isinstance(node, exp.Column):
       predicate_name = self._get_predicate_name(predicate_count)
       predicate_mappings[predicate_name] = node
       return predicate_name, predicate_count + 1
     else:
       raise NotImplementedError
 
 
-  def _get_original_predicate(self, predicate_name, predicate_mappings) -> FilteringPredicate:
-    if predicate_name[0] == "~":
-      return FilteringPredicate(True, predicate_mappings[predicate_name[1:]])
-    else:
-      return FilteringPredicate(False, predicate_mappings[predicate_name])
-
-
-  def _get_or_clause_representation(self, or_expression, predicate_mappings) -> List[FilteringPredicate]:
+  def _get_or_clause_representation(self, or_expression, predicate_mappings):
     connected_by_ors = list(or_expression.args)
     predicates_in_ors = []
     if len(connected_by_ors) <= 1:
-      predicates_in_ors.append(self._get_original_predicate(str(or_expression), predicate_mappings))
+      predicates_in_ors.append(predicate_mappings[str(or_expression)])
```

Comment:
Do the expressions always normalize to the same thing, e.g., if you have "A OR B" vs "B OR A"? Or does that different not matter here

Reply:
The order doesn't matter.